backend/app/model/digital_object_component.rb

```diff
@@ -88,4 +88,15 @@ def self.touch_records(obj)
     [{ type: DigitalObject, ids: [obj.root_record_id] }]
   end
 
+  def delete
+    affected_resource_ids = DigitalObject.instance_owners_root_records(self[:root_record_id])
+    Resource.update_mtime_for_ids(affected_resource_ids)
+    super
+  end
+
+  def update_from_json(json, opts = {}, apply_nested_records = true)
+    affected_resource_ids = DigitalObject.instance_owners_root_records(self[:root_record_id])
+    Resource.update_mtime_for_ids(affected_resource_ids)
+    super
+  end
```

Comment:
These methods could be replaced with an addition to `touch_records` I think:

```ruby
  def self.touch_records(obj)
    [
      { type: DigitalObject, ids: [obj.root_record_id] },
      { type: Resource, ids: DigitalObject.instance_owners_root_records(obj.root_record_id) }
    ]
  end
```

And `touch_records` could be added to digital_object.rb for the equivalent change.

However I think you looked at `touch_records` so may want to avoid it. Also I don't think it (touch) can address the order of operations for bumping mtimes _after_ delete, per feedback above, because it receives the object before it's deleted (touch records should instead be working with the current record's id to support all cases, including after delete -- but it doesn't make sense to look at that now).

Reply:
I did look at it and thought for some reason that it would not work for this case, but it seems lik your suggestion does work fine, so I did make this change. I also made a small change to the`TouchRecords` mixin on the assumption that it will always be preferable to delete first, and then update related mtimes.